modules/spring-config/src/main/java/org/mule/runtime/config/spring/MuleArtifactContext.java

```diff
@@ -294,6 +299,11 @@ protected void loadBeanDefinitions(DefaultListableBeanFactory beanFactory) throw
                                                                                   {
                                                                                       nameAttribute = OBJECT_MULE_CONFIGURATION;
                                                                                   }
+                                                                                  else if (nameAttribute == null)
+                                                                                  {
+                                                                                      //This may be a configuration that does not requires a name.
+                                                                                      nameAttribute = uniqueValue(resolvedComponentModel.getBeanDefinition().getBeanClassName());
```

Comment:
does this validate that no top level element is without a name? is this a fix for MULE-10097?